src/components/CasdoorCard/index.js

```diff
@@ -1,20 +1,74 @@
-import React from "react";
-import {useEffect, useState} from "react";
+import Translate from "@docusaurus/Translate";
+import Link from "@docusaurus/Link";
+import React, {useEffect, useState} from "react";
+import axios from "axios";
 
 export default function CasdoorCard(props) {
+  const [loading, setLoading] = useState(false);
   const [link, setLink] = useState("https://door.casdoor.org/");
+  const [mouseState, setMouseState] = useState({state: false});
+  const maskStyle = {
+    position: "absolute",
+    top: "0px",
+    left: "0px",
+    zIndex: "1",
+    height: "100%",
+    width: "100%",
+    background: "rgba(0,0,0,0.4)",
+    cursor: "pointer",
+    borderRadius: "20px",
+  };
   useEffect(() => {
-    setLink(localStorage.getItem("CasdoorLink"));
+    axios.get("https://oa.casbin.com/api/is-mainland-ip").then((response) => {
+      if (response.data === true) {
+        setLink("https://door.casdoor.com/");
+      } else {
+        setLink("https://door.casdoor.org/");
+      }
+    });
+    setLoading(true);
   }, []);
+
+  const handleMouseEnter = () => {
+    setMouseState({
+      ...mouseState,
+      state: true,
+    });
+
+  };
+  const handleMouseLeave = () => {
+    setMouseState({
+      ...mouseState,
+      state: false,
+    });
+  };
   return (
-    <iframe
-      src={link + props.src}
-      width={props.width}
-      height={props.height}
-      style={{borderRadius: "20px"}}
-      frameBorder="0"
-      scrolling="no"
-    ></iframe>
+    <div className={props.className} onMouseEnter={handleMouseEnter} onMouseLeave={handleMouseLeave}>
+      <iframe
+        src={link + props.src}
+        width={props.width}
+        height={props.height}
+        style={{borderRadius: "20px"}}
+        frameBorder="0"
+        scrolling="no"
+      />
+      <div style={maskStyle} hidden={!mouseState.state || !props.mask} onClick={() => {window.open(link + props.src);}}>
+        <Link
+          className="button button--secondary button--lg"
+          style={{marginTop: "50%", marginRight: "3rem", marginLeft: "3rem"}}>
+          <Translate>Online Demo</Translate>
+        </Link>
+      </div>
+      <div hidden={loading} style={{
```

Comment:
![image](https://user-images.githubusercontent.com/53544726/190268892-a1027961-93e6-4975-8de4-2320c3a40cbc.png)

![image](https://user-images.githubusercontent.com/53544726/190268996-91edee0b-d57f-4a17-8e88-ce4d11c21f39.png)

In https://deploy-preview-339--iridescent-baklava-0d7550.netlify.app/docs/overview
I removed the `hidden` attribute to see the effect, and found that it was not good

Reply:
Right, I have resolved it. plz review